pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1045,19 +1047,26 @@ public CompletableFuture<Optional<Topic>> getTopic(final TopicName topicName, bo
             }
             final boolean isPersistentTopic = topicName.getDomain().equals(TopicDomain.persistent);
             if (isPersistentTopic) {
-                return topics.computeIfAbsent(topicName.toString(), (tpName) -> {
-                    if (topicName.isPartitioned()) {
-                        return fetchPartitionedTopicMetadataAsync(TopicName.get(topicName.getPartitionedTopicName()))
-                                .thenCompose((metadata) -> {
-                                    // Allow crate non-partitioned persistent topic that name includes `partition`
-                                    if (metadata.partitions == 0
-                                            || topicName.getPartitionIndex() < metadata.partitions) {
-                                        return loadOrCreatePersistentTopic(tpName, createIfMissing, properties);
-                                    }
-                                    return CompletableFuture.completedFuture(Optional.empty());
-                                });
-                    }
-                    return loadOrCreatePersistentTopic(tpName, createIfMissing, properties);
+                final CompletableFuture<Optional<TopicPolicies>> topicPoliciesFuture =
+                        getTopicPoliciesBypassSystemTopic(topicName);
+                return topicPoliciesFuture.thenCompose(optionalTopicPolicies -> {
```

Comment:
Not sure if the broker can handle all the exceptions from the topic policies service when loading topics. It's better to check to make sure we will not return any unexpected errors to the client.

Reply:
fixed.